gremlin-go/driver/traversal.go

```diff
@@ -120,3 +120,258 @@ func (t *Traversal) getResults() (ResultSet, error) {
+	// And Predicate returns a Predicate composed of two predicates (logical AND of them).
+	And(args ...interface{}) Predicate
+	// Or Predicate returns a Predicate composed of two predicates (logical OR of them).
+	Or(args ...interface{}) Predicate
+}
+
+var P *p = &p{}
+
+func newP(operator string, args ...interface{}) Predicate {
+	values := make([]interface{}, 0)
+	for _, arg := range args {
+		values = append(values, arg)
+	}
+	return &p{operator: operator, values: values}
+}
+
+func newPWithP(operator string, pp p, args ...interface{}) Predicate {
+	values := make([]interface{}, len(args)+1)
+	for _, arg := range args {
+		values = append(values, arg)
+	}
+	values[len(values)-1] = pp
+	return &p{operator: operator, values: values}
+}
+
+func (_ *p) Between(args ...interface{}) Predicate {
+	return newP("between", args...)
+}
+
+func (_ *p) Eq(args ...interface{}) Predicate {
+	return newP("eq", args...)
+}
+
+func (_ *p) Gt(args ...interface{}) Predicate {
+	return newP("gt", args...)
+}
+
+func (_ *p) Gte(args ...interface{}) Predicate {
+	return newP("gte", args...)
+}
+
+func (_ *p) Inside(args ...interface{}) Predicate {
+	return newP("inside", args...)
+}
+
+func (_ *p) Lt(args ...interface{}) Predicate {
+	return newP("lt", args...)
+}
+
+func (_ *p) Lte(args ...interface{}) Predicate {
+	return newP("lte", args...)
+}
+
+func (_ *p) Neq(args ...interface{}) Predicate {
+	return newP("neq", args...)
+}
+
+func (_ *p) Not(args ...interface{}) Predicate {
+	return newP("not", args...)
+}
+
+func (_ *p) Outside(args ...interface{}) Predicate {
+	return newP("outside", args...)
+}
+
+func (_ *p) Test(args ...interface{}) Predicate {
+	return newP("test", args...)
+}
+
+func (_ *p) Within(args ...interface{}) Predicate {
+	return newP("within", args...)
+}
+
+func (_ *p) Without(args ...interface{}) Predicate {
+	return newP("without", args...)
+}
+
+func (pp *p) And(args ...interface{}) Predicate {
+	return newPWithP("and", *pp, args...)
+}
+
+func (pp *p) Or(args ...interface{}) Predicate {
+	return newPWithP("or", *pp, args...)
+}
+
+type TextPredicate interface {
+	// Containing TextPredicate determines if a string contains a given value.
+	Containing(args ...interface{}) TextPredicate
+	// EndingWith TextPredicate determines if a string ends with a given value.
+	EndingWith(args ...interface{}) TextPredicate
+	// NotContaining TextPredicate determines if a string does not contain a given value.
+	NotContaining(args ...interface{}) TextPredicate
+	// NotEndingWith TextPredicate determines if a string does not end with a given value.
+	NotEndingWith(args ...interface{}) TextPredicate
+	// NotStartingWith TextPredicate determines if a string does not start with a given value.
+	NotStartingWith(args ...interface{}) TextPredicate
+	// StartingWith TextPredicate determines if a string starts with a given value.
+	StartingWith(args ...interface{}) TextPredicate
+}
+
+type textP p
+
+var TextP *textP = &textP{}
```

Comment:
This should be var TextP Textpredicate = &textP{}

Reply:
```suggestion
var TextP Textpredicate = &textP{}
```